pulsar/producer_partition.go

```diff
@@ -299,6 +278,26 @@ func (p *partitionProducer) grabCnx() error {
 	if err != nil {
 		return err
 	}
+
+	if p.options.DisableBatching {
```

Comment:
The following code looks like more clear:
```
	if !p.options.DisableBatching {
		batcherBuilderType := p.options.BatcherBuilderType
		provider, err := GetBatcherBuilderProvider(batcherBuilderType)
		if err != nil {
			return err
		}
		maxMessageSize := uint32(p._getConn().GetMaxMessageSize())
		p.batchBuilder, err = provider(p.options.BatchingMaxMessages, p.options.BatchingMaxSize,
			maxMessageSize, p.producerName, p.producerID, pb.CompressionType(p.options.CompressionType),
			compression.Level(p.options.CompressionLevel),
			p,
			p.log,
			p.encryptor)
		if err != nil {
			return err
		}
	}
```

Reply:
Good suggestion, thx.